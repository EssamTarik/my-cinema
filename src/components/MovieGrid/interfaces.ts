import { IMovie } from "../../store/movieSearch/models";
import { IFavorites } from "../../store/favorites/models";
import { IWatchLater } from "../../store/watchLater/models";

export interface IProps {
  movies: IMovie[];
  favorites: IFavorites;
  watchLater: IWatchLater;
  onAddToFavorites: (movie: IMovie) => void;
  onRemoveFromFavorites: (movie: IMovie) => void;
  onAddToWatchLater: (movie: IMovie) => void;
  onRemoveFromWatchLater: (movie: IMovie) => void;
}