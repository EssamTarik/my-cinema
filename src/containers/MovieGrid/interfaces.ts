import { IMovie } from "../../store/movieSearch/models";

export interface IProps {
  movies: IMovie[];
}