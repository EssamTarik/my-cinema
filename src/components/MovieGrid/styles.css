.movie-grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  margin-bottom: 20px;
}

.movie-grid > div {
  margin: 20px 20px;
}