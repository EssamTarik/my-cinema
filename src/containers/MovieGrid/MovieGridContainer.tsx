import React from 'react';
import { IProps } from './interfaces';
import { MovieGrid } from '../../components';

const MovieGridContainer = ({ movies, favorites, addFavorite, removeFavorite }: IProps) => (
  <MovieGrid
    favorites={favorites}
    onAddToFavorites={addFavorite}
    onRemoveFromFavorites={removeFavorite}
    movies={movies}
  />
);

export default MovieGridContainer;