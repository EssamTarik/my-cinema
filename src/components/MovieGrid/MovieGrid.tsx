import React from 'react';
import MovieCard from '../MovieCard';
import { IProps } from './interfaces';
import './styles.css';
import { FormattedMessage } from 'react-intl';

const MovieGrid = ({ movies, favorites, onAddToFavorites, onRemoveFromFavorites }: IProps) => {
  if (movies.length === 0) {
    return <div className='movie-grid__no-results'><FormattedMessage id='movie.noResults' /></div>
  }
  return (
    <div className='movie-grid'>
      {
        movies.map(movie => {
          const isFavorite = favorites[movie.id] !== undefined;
          return (
            <MovieCard
              key={movie.id}
              movie={movie}
              favorite={isFavorite}
              watchLater={true}
              onAddToFavorites={() => onAddToFavorites(movie)}
              onAddToWatchLater={console.log}
              onRemoveFromFavorites={() => onRemoveFromFavorites(movie)}
              onRemoveFromWatchLater={console.log}
            />
          )
        })
      }
    </div>
  )
}

export default MovieGrid;