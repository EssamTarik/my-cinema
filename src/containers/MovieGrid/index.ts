import { connect } from 'react-redux';
import { addFavorite, removeFavorite } from '../../store/favorites/actions';
import MovieGridContainer from './MovieGridContainer';
import { IStateTree } from '../../store/interfaces';

const mapDispatchToProps = {
  addFavorite,
  removeFavorite
};

const mapStateToProps = ({ favorites }: IStateTree) => ({ favorites });

export default connect(mapStateToProps, mapDispatchToProps)(MovieGridContainer);