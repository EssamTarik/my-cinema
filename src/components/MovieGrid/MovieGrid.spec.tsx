jest.mock('../MovieCard', () => () => <div />)
jest.mock('react-intl', () => ({ FormattedMessage: () => <div /> }));
import React from "react";
import { render, cleanup } from "@testing-library/react";

const movieMock = jest.requireMock('movie');

import MovieGrid from "./MovieGrid";

describe('MovieGrid', () => {
  afterEach(cleanup);
  it("renders", () => {
    const { asFragment } = render(
      <MovieGrid favorites={{}} onAddToFavorites={jest.fn()} onRemoveFromFavorites={jest.fn()} movies={[movieMock]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('returns no results message', () => {
    const { container } = render(
      <MovieGrid favorites={{}} onAddToFavorites={jest.fn()} onRemoveFromFavorites={jest.fn()} movies={[]} />
    );
    const noResultsElement = container.querySelector('.movie-grid__no-results');
    expect(noResultsElement).not.toEqual(null);
  })
})

