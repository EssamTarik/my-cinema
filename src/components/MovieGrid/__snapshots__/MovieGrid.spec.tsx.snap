// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGrid renders 1`] = `
<DocumentFragment>
  <div
    class="movie-grid"
  >
    <div />
  </div>
</DocumentFragment>
`;
